xcode/Subconscious/Shared/Services/Noosphere/Sphere.swift

```diff
@@ -158,13 +155,35 @@ public actor Sphere: SphereProtocol, SpherePublisherProtocol {
     public let sphere: OpaquePointer
     private let _identity: String
     
-    private init(noosphere: Noosphere, identity: String, sphere: OpaquePointer) throws {
+    /// Initialize a sphere from an already-created sphere pointer, retrieving
+    /// identity from Noosphere.
+    ///
+    /// This is a private initializer used in implementing traversal.
+    private init(
```

Comment:
Create private intializer that calls `ns_sphere_identity` to read identity from already-initialized sphere pointer.

Note: initializers don't support async, so initialization must always be synchronous in Swift. Implementing this private init this way is based on the assumption that as of Noosphere 0.9.x, reading sphere identity from pointer will have minimal latency (will not be blocked on writes, and will not reach out to the network).

Note however that we do only call this initializer from the async method `traverse` (https://github.com/subconsciousnetwork/subconscious/pull/514/files#diff-6deb9e69a5a4545d3d22efbc1696a7f88ab6dc55137b68618eca8063aad0f1ccR605). Calling even a blocking initializer in that context will never block the main thread (if Swift's threadpool is doing its job).

Reply:
This is reasonable. This call should never reach out to the network or block for a long period of time.